code/controllers/subsystems/inactivity.dm

```diff
@@ -1,11 +1,12 @@
-/datum/controller/process/inactivity/setup()
-	name = "inactivity"
-	schedule_interval = 600 // Once every minute (approx.)
+SUBSYSTEM_DEF(inactivity)
+	name = "AFK Kick"
+	wait = 600
+	flags = SS_BACKGROUND | SS_NO_TICK_CHECK
 
-/datum/controller/process/inactivity/doWork()
+/datum/controller/subsystem/inactivity/fire()
 	if(config.kick_inactive)
-		for(last_object in clients)
-			var/client/C = last_object
+		for(var/i in clients)
+			var/client/C = i
```

Comment:
It's allegedly faster to do it that way since I think ``for(var/client/C in clients)`` does a typecheck for each iteration, where as this way doesn't do that but you need to be absolutely sure the list is homogeneous.

Reply:
what neerti said was what i was told.